server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java

```diff
@@ -23,14 +23,17 @@
  * @opensearch.internal
  */
 public class RegisterTransportActionsRequest extends TransportRequest {
+    private String uniqueId;
     private Map<String, Class> transportActions;
 
-    public RegisterTransportActionsRequest(Map<String, Class> transportActions) {
+    public RegisterTransportActionsRequest(String uniqueId, Map<String, Class> transportActions) {
+        this.uniqueId = uniqueId;
         this.transportActions = new HashMap<>(transportActions);
     }
 
     public RegisterTransportActionsRequest(StreamInput in) throws IOException {
         super(in);
+        this.uniqueId = in.readString();
```

Comment:
I'm glad to see the uniqueId being passed around but not clear what we're doing with it yet. :)

Reply:
Haha yeah, we should just follow one way of identification.
Its used to look up the actions registered by that extension[1]

[1] https://github.com/opensearch-project/OpenSearch/pull/4598/files#diff-975beb295a2ff9de0a0b4a5df61ea60f5ea4fddb7f1c801e1161860041446238R75